doc/minimal-manual.nix

```diff
@@ -0,0 +1,41 @@
+# Minimal manual prototype
+
+with import <nixpkgs> {};
+
+let
+  rootPath = "http://localhost:8000/result/";
+
+  addRootPath = '' ${gnused}/bin/sed -r 's|<a href="/|<a href="${rootPath}|g' '';
+  addHeaderAnchors = '' ${gnused}/bin/sed -r 's/# ([^{]*)\s\{#([^}]*)}/# <span id="\2">[\1](#\2)<\/span>/g' '';
+  addAdmonitions = '' ${gnused}/bin/sed -r 's/::: warning/<div class="warning"><h3 class="title">Warning<\/h3><p>/g' | ${gnused}/bin/sed -r 's/:::/<\/p><\/div>/g' '';
+  cmarkCmd = "${cmark-gfm}/bin/cmark-gfm --unsafe";
+  cmd = "${addAdmonitions} | ${addHeaderAnchors} | ${cmarkCmd} | ${addRootPath}";
+  htmlHead = "<!doctype html><html><head><meta charset=\\\"utf-8\\\"><link href=\\\"${rootPath}minimal.css\\\" rel=\\\"stylesheet\\\" type=\\\"text/css\\\" /></head><body>";
+  htmlFoot = "</body>";
+in
+stdenv.mkDerivation {
+  name = "nixpkgs-minimal-manual";
+  src = builtins.filterSource (path: type: type == "directory" || builtins.match ".*\.md" path == []) ./.;
+  builder = writeScript "cmark-builder" ''
+    PATH=${coreutils}/bin
+    mkdir -p $out
+    cp "${./minimal.css}" "$out/minimal.css"
+    toc="$(cat "$src/toc.md" | ${cmd})"
+    for file in $src/*.md $src/**/*.md; do
+      echo $file
+      outputFile="''${file%.md}"
+      outputFile="''${outputFile%.section}"
+      outputFile="''${outputFile%.chapter}"
+      outputFile="$out/''${outputFile#"$src/"}.html"
+      mkdir -p "$(dirname "$outputFile")"
+      echo "${htmlHead}" >> "$outputFile"
+      if [ $file != "$src/toc.md" ]; then
+        echo "$toc" >> "$outputFile"
+      fi
+      echo "<section>" >> "$outputFile"
+      cat "$file" | ${cmd} >> "$outputFile"
+      echo "</section>" >> "$outputFile"
+      echo "${htmlFoot}" >> "$outputFile"
```

Comment:
some here-doc would be more readable here.

Reply:
Yep, it did, thanks.